src/main/java/com/google/sps/data/UserInfo.java

```diff
@@ -28,29 +31,32 @@
   public static final String BOOKMARKED_EVENTS = "bookmarked_events";
   public static final String CURRENT_CHALLENGE = "current_challenge";
   public static final String CHALLENGE_STATUSES = "challenge_statuses";
+  public static final String ENTITY_KEY = "entity_key";
 
   private final String id;
   private final String nickname;
   private final List<Long> created_events;
   private final List<Long> bookmarked_events;
   private Long current_challenge_id;
   private List<Integer> challenge_statuses;
+  private Key entity_key;
 
-
-  public UserInfo(String id, String nickname) {
+  public UserInfo(String id, String nickname, Key entity_key) {
```

Comment:
[No change requested]: You should read about the builder pattern, which can help remove some of these constructors with extra long parameter lists (https://dzone.com/articles/design-patterns-the-builder-pattern).